### PR TITLE
Blob java

### PR DESCRIPTION
**Pull request by kirkeaton**
1. Added java generation of Blob, BlobService, AbstractBlob, ProxyBlob
2. Added IdentifiedBlob

### CODE REVIEW

File: src/foam/blob/BlobJava.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        },
+        {
+          name: 'id',
+          javaType: 'String'
+        }
+      ]
+    },
+    {
+      name: 'urlFor',
+      javaReturns: 'String[]',
```
Comment (adamvy) [inline src/foam/blob/BlobJava.js]: Not String[], just String

Comment (kirkeaton) [reply src/foam/blob/BlobJava.js]: The JS version has `returns: 'Strings'` is that a typo then?

File: src/foam/blob/IdentifiedBlob.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ * limitations under the License.
+ */
+
+package foam.blob;
+
+import foam.core.X;
+
+import java.nio.Buffer;
+
+public class IdentifiedBlob
```
Comment (adamvy) [inline src/foam/blob/IdentifiedBlob.java]: We should generate this from the foam.CLASS so that serialization works.

Comment (kirkeaton) [reply src/foam/blob/IdentifiedBlob.java]: IdentifiedBlob has a property called delegate with a factory that looks up the delegate based on the ID. How would I handle that when auto generating?

File: src/foam/blob/BlobJava.js (hunk 1 above)
Comment (adamvy) [reply src/foam/blob/BlobJava.js]: Yeah that's a typo

File: src/foam/blob/IdentifiedBlob.java (hunk 2 above)
Comment (adamvy) [reply src/foam/blob/IdentifiedBlob.java]: Add a javaFactory to the property to do the same thing

**Pull request merged**